etna/datasets/tsdataset.py

```diff
@@ -656,16 +656,17 @@ def to_dataset(df: pd.DataFrame) -> pd.DataFrame:
         2021-01-04           3           8
         2021-01-05           4           9
         """
-        df["timestamp"] = pd.to_datetime(df["timestamp"])
-        df["segment"] = df["segment"].astype(str)
-        feature_columns = df.columns.tolist()
+        df_copy = df.copy(deep=True)
+        df_copy["timestamp"] = pd.to_datetime(df["timestamp"])
```

Comment:
It's better to use `df_copy` at the right  and below too

Reply:
Done